### DISCUSSION TITLE
https://1kevinson.com/upload-and-download-image-into-sql-database-with-spring-boot/

### DISCUSSION THREAD

**Discussion by giscus** [1kevinson/BLOG-TUTORIALS#14] [Announcements]
# https://1kevinson.com/upload-and-download-image-into-sql-database-with-spring-boot/

By the end of this article, you will have an understanding of how to implement an image upload and download feature using Spring Boot.

https://1kevinson.com/upload-and-download-image-into-sql-database-with-spring-boot/

**Comment by LiveNathan** [comment]
Thanks for this Kevin. I'm using Hibernate 6 so `@Type(type = "org.hibernate.type.ImageType")` doesn't work anymore. Do you know the solution?

**Reply by 1kevinson** [reply]
Hi @LiveNathan, I don't know the exact type which replace the hibernate 3 one.
But could you try the `AnyType`  from the [Hibernate 6 official docs](https://docs.jboss.org/hibernate/orm/6.2/javadocs/org/hibernate/type/AnyType.html), and let me know if it resolve your issue ?

**Comment by Tomasus-prog** [comment]
Thanks for your work, it's very helpful.  I'm student and in my project I have to create solution on adding image to userProfile. Principles my project is userProfile can have image, but not always, so Can I connect Tables user_profile and image, when image_id is Foreign Key in user_profile table, and then your solution works fine?

**Reply by 1kevinson** [reply]
Hello @Tomasus-prog, yes, the solution in this article should work. Just retrieve the image byte array from your table  by finding it by ID.
